Prog/Affordability_sf_condo.sas

```diff
@@ -0,0 +1,321 @@
+ Library:  DMPED
+ Project:  NeighborhoodInfo DC
+ Author:   Yipeng Su
+ Created:  6/28/18
+ Version:  SAS 9.4
+ Environment:  Windows with SAS/Connect
+ 
+ Description: *Methodology for affordability adapted from Zhong Yi Tong paper 
+http://content.knowledgeplex.org/kp2/cache/documents/22736.pdf
+Homeownership Affordability in Urban America: Past and Future;
+ Modifications: 
+**************************************************************************/
+
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+
+** Define libraries **;
+%DCData_lib( DMPED );
+%DCData_lib( equity );
+%DCData_lib( realprop );
+
+%Sales_pars();
+%Clean_sales( inds=DMPED.Sales_who_owns_SF_Condo, outds=Sales_who_owns_SF_Condo_clean) ;
+
+proc contents data= Sales_who_owns_SF_Condo_clean; run;
+
+data create_flags;
+  set Sales_who_owns_SF_Condo_clean (where=(LargeUnit=1 and clean_sale=1 and (2000 <= saleyear <= 2017)));
+  
+  /*pull in effective interest rates - for example: 
+  http://www.fhfa.gov/DataTools/Downloads/Documents/Historical-Summary-Tables/Table15_2015_by_State_and_Year.xls*/
+  
+    eff_int_rate_2000= 7.90;
+	eff_int_rate_2001= 7.07;
+	eff_int_rate_2002= 6.62;
+	eff_int_rate_2003= 5.73;
+	eff_int_rate_2004= 5.53;
+	eff_int_rate_2005= 5.67;
+	eff_int_rate_2006= 6.53;
+	eff_int_rate_2007= 6.54;
+	eff_int_rate_2008= 6.14;
+	eff_int_rate_2009= 5.02;
+    eff_int_rate_2010= 4.93;
+	eff_int_rate_2011= 4.62;
+	eff_int_rate_2012= 3.72;
+	eff_int_rate_2013= 3.95;
+	eff_int_rate_2014= 4.22;
+	eff_int_rate_2015= 3.95;
+	eff_int_rate_2016= 3.69;
+    eff_int_rate_2017= 3.69; *2017 not available, using 2016;
+	%macro yearloop ();
+
+	%do year = 2000 %to 2017 ;
+
+	month_int_rate_&year. = (eff_int_rate_&year./12/100);
+
+	loan_multiplier_&year. =  month_int_rate_&year. *	( ( 1 + month_int_rate_&year. )**360	) / ( ( ( 1+ month_int_rate_&year. )**360 )-1 );
+
+
+	*calculate monthly Principal and Interest for First time Homebuyer (10% down);
+	if saleyear=&year. then PI_First&year.=saleprice*.9*loan_multiplier_&year.;
+
+	%dollar_convert(PI_first&year.,PI_first&year.r,&year.,2017);
+
+   *calculate monthly PITI (Principal, Interest, Taxes and Insurance) for First Time Homebuyer (34% of PI = TI);
+	if saleyear=&year. then PITI_First=PI_First&year.r*1.34;
+
+  *calculate monthly Principal and Interest for Repeat Homebuyer (20% down);
+    if saleyear=&year. then PI_Repeat&year.=saleprice*.8*loan_multiplier_&year.;
+
+	%dollar_convert(PI_Repeat&year.,PI_Repeat&year.r,&year.,2017);
+
+*calculate monthly PITI (Principal, Interest, Taxes and Insurance) for Repeat Homebuyer (25% of PI = TI);
+	if saleyear=&year. then PITI_Repeat=PI_Repeat&year.r*1.25;
+
+	%end;
+
+	%mend yearloop;
+	%yearloop
+
+
+	/*Here are numbers for Average Household Income at the city level. 2012-16 ACS 
+Black	NH-White	Hispanic	AIOM	 
+61923	 165970 	92543 	 	 80028		
+numhshldsb_2012_16 numhshldsw_2012_16 numhshldsh_2012_16 numhshldsaiom_2012_16*/ 
+
+	/*80%AMI and 50% AMI as the new affordability category  
```

Comment:
@maggiesu0725 - okay to leave as is but you could have done a loop with the years here too.